go.mod

```diff
@@ -6,10 +6,9 @@ require (
 	github.com/appscode/go v0.0.0-20200323182826-54e98e09185a
 	github.com/getsentry/sentry-go v0.4.0
 	github.com/golang/mock v1.6.0
-	github.com/linode/linodego v0.32.2
-	github.com/pkg/errors v0.9.1
+	github.com/linode/linodego v1.25.0
```

Comment:
Linodego is upgraded significantly here, but not in [e2e/go.mod](https://github.com/rammanoj/linode-cloud-controller-manager/blob/56d93e8b98f38e30c94c2820b36fcf30efaa15d8/e2e/go.mod#L9C17-L9C17). Is there any reason for this mismatch?

Reply:
I missed upgrading it. I'll work on it. Thanks!

Reply:
The existing e2e tests are already breaking! When I tried executing them without any changes from existing PR, these are breaking! The terraform is not able to find `python` on one of the linodes!
PFA image and the detailed log of the tests.
![image](https://github.com/linode/linode-cloud-controller-manager/assets/21082378/9e91f156-794d-4a05-9eb5-b218998f487b)
[log.txt](https://github.com/linode/linode-cloud-controller-manager/files/13503877/log.txt)

Since these are already breaking. Fixing these tests and upgrading the linodego in it could be taken on a separate PR.